backend/controllers/movieController.js

```diff
@@ -27,8 +29,44 @@ const getMovieEntry = async (req, res, next) => {
   } catch (err) { next(err); }
 };
 
+const addComment = async (req, res, next) => {
+  try {
+    const newComment = {
+      userId: req.body.userId,
+      comment: req.body.comment,
+    };
+    const imdbCode = req.params.imdb_code;
+
+    if (newComment.comment === '') throw detailedError();
+
+    const movieId = await Movie.findOneAndUpdate({
+      _id: imdbCode,
+    }, {
+      $addToSet: {
+        comments: newComment,
+      },
+    });
+
+    if (movieId === null) {
+      const newMovieInDB = new Movie({
+        _id: imdbCode,
```

Comment:
Did this work? 
Because I tried to create a comment, and mongoose threw me a CastError, that _id  Argument passed in must be a single string of 12 bytes or a string of 24 hex characters. It seems like mongoose ids are hex characters.
Should we add another property to our Movie, like _imdbCode_, that will be unique?

Reply:
Yes, it needs to be a hex number, because of Mongodb's index system and its the only way how I could find the movie and push the new comment to it with the parameters.
I agree, maybe we do need a new field in the Schema.

Reply:
> I agree, maybe we do need a new field in the Schema.
Certainly seems like it, such a shame but doesn't really change the grand scheme of things too much!